### PR TITLE
Integrate Record Skipping into Column Reader Fuzz Test

### LINKED ISSUE #2198: Integrate Record Skipping into Column Reader Fuzz Tests
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

`parquet::arrow::arrow_reader` contains a fairly extensive set of tests under `run_single_column_reader_tests` that are used to fuzz test the arrow readers. It would significantly boost the coverage of the record skipping logic if we could integrate the row skipping logic into this harness

**Describe the solution you'd like**

Add an optional row selection to `TestOptions` and add examples of using this to the `all_options` list in `run_single_column_reader_tests`.

**Describe alternatives you've considered**

We could not do this

**Additional context**

- #2197 
- #1792

FYI @Ted-Jiang

### PR DESCRIPTION
**Pull request by Ted-Jiang**
# Which issue does this PR close?

Closes #2198.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by Ted-Jiang** [comment]
Got 4 test error 😂 will dive into
image

**Comment by Ted-Jiang** [comment]
There are two kinds of erroe
one is 
```
failed to skip rows, expected x, got 4
```

one is  parquet/src/arrow/arrow_reader.rs:1260
```
assert_eq!(a.data(), b.data(), "{:#?} vs {:#?}", a.data(), b.data());
```

**Comment by Ted-Jiang** [comment]
as test `test_primitive_single_column_reader_test`
it fail at `write_batch_size: 64, max_data_page_size: 128, max_dict_page_size: 1048576, writer_version: PARQUET_1_0, enabled_statistics: Page, encoding: PLAIN, row_selections: Some` INT96  version i will check this first 🤔

**Comment by tustvold** [comment]
Awesome! FWIW it might help debugging to seed the random number generator to get a deterministic test. I've found the fuzz tests are very good at finding bugs, but can be quite hard to then debug 😅

**Comment by Ted-Jiang** [comment]
> Awesome! FWIW it might help debugging to seed the random number generator to get a deterministic test. I've found the fuzz tests are very good at finding bugs, but can be quite hard to then debug 😅

Thanks !  i have found the combination args cause bug 😂

**Comment by liukun4515** [comment]
> ConvertedType::NONE/ArrowType::None

From the error message, I guess the error is related to the `ConvertedType::NONE/ArrowType::None`.

read/convert parquet data to arrow, we need the target arrow type and parquet logical type/convertedtype

**Comment by Ted-Jiang** [comment]
@tustvold  seems all tests passed.  I copy the all test case with page static. 😊

File: parquet/src/arrow/arrow_reader.rs [hunk 1]
```diff
@@ -1126,36 +1178,49 @@ mod tests {
...
+            let options =
+                ArrowReaderOptions::new().with_row_selection(selections.clone());
+            arrow_reader =
+                ParquetFileArrowReader::try_new_with_options(file, options).unwrap();
+            let mut without_skip_data = gen_expected_data::<T>(&def_levels, &values);
+
+            let mut skip_data: Vec<Option<T::T>> = vec![];
+            for select in selections {
+                if select.skip {
+                    without_skip_data.drain(0..select.row_count);
```
Comment (tustvold) [inline parquet/src/arrow/arrow_reader.rs:1193]: Nice :ok_hand:

File: parquet/src/arrow/arrow_reader.rs [hunk 2]
```diff
@@ -1126,36 +1178,49 @@ mod tests {
...
         let mut total_read = 0;
         loop {
             let maybe_batch = record_reader.next();
             if total_read < expected_data.len() {
-                let end = min(total_read + opts.record_batch_size, expected_data.len());
+                let mut end =
+                    min(total_read + opts.record_batch_size, expected_data.len());
                 let batch = maybe_batch.unwrap().unwrap();
-                assert_eq!(end - total_read, batch.num_rows());
+                //TODO remove this after implement https://github.com/apache/arrow-rs/issues/2197
```
Comment (tustvold) [inline parquet/src/arrow/arrow_reader.rs:1218]: The linked issue has been done?

**Review by tustvold — APPROVED** [review]
Very nice :+1:

File: parquet/src/arrow/arrow_reader.rs (hunk 2 above)
Comment (Ted-Jiang) [reply parquet/src/arrow/arrow_reader.rs:1218]: Nope, maybe merge this and use this test to check the result

**Pull request merged**